### PR TITLE
[Exafmmt] New package

### PR DESCRIPTION
**Pull request by JoshuaTetzner**
Created a build script for Exafmm-t. Since the library uses some unix only functions, a windows build is not available.

### CODE REVIEW

File: E/Exafmmt/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"); platforms=filter(!Sys.isbsd, platforms))
+    Dependency(PackageSpec(name="LLVMOpenMP_jll", uuid="1d63c593-3942-5779-bab2-d838dc0a180e"); platforms=filter(Sys.isbsd, platforms))
+    Dependency(PackageSpec(name="OpenBLAS_jll", uuid="4536629a-c528-5b80-bd46-f80d51c5b363"))
+    Dependency(PackageSpec(name="LAPACK_jll", uuid="51474c39-65e3-53ba-86ba-03b1b862ec14"))
+    Dependency(PackageSpec(name="FFTW_jll", uuid="f5851436-0d7a-5f13-b9de-f02708fd171a"))
+
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"12.1.0")
```
Comment (giordano) [inline E/Exafmmt/build_tarballs.jl:63]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"12.1.0") 

```

File: E/Exafmmt/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+#/workspace/x86_64-linux-gnu-libgfortran5-cxx11/artifacts/9f61595dd5f6597db74a071ac26c2d094b563029/lib/libblastrampoline.so
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd exafmm-t/
+mkdir build
+cd build
+BLAS=blastrampoline
+LAPACK=blastrampoline
```
Comment (giordano) [inline E/Exafmmt/build_tarballs.jl:21]: You don't have libblastrampoline in your dependencies.  Instead, you have OpenBLAS and LAPACK which I don't understand what they're used for.

**Comment by giordano** [comment]
@JoshuaTetzner you don't have to close PRs every time, things can be worked out on the same PR.  Closing PRs splits reviews over multiple pages, which makes it hard to follow them.  For future reference, last revision before this branch was cleared out is https://github.com/JuliaPackaging/Yggdrasil/commit/8cf06ca13e9b609a5c1f732f0800fcceaee0a2bc

**Comment by JoshuaTetzner** [comment]
Sorry for that. I reset the branch to the main to clean the commit history, but pushed it before commiting again which closed the pull request automatically.
